### PR TITLE
feat(providers): add threads

### PR DESCRIPTION
**Pull request by Lqm1**
## ☕️ Reasoning

Now that the threads.net API is out, I'll add it to the provider.

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Comment by ndom91** [comment]
Nice, thanks! 

A docs page under `/docs/pages/getting-started/providers/threads.mdx` would be great 🙏 just check out any of the other existing ones for an example

**Comment by c7e715d1b04b17683718fb1e8944cc28** [comment]
Okay! Add now

**Comment by ThangHuuVu** [comment]
@ndom91 could you help to setup a Thread OAuth app? 👀

**Comment by ndom91** [comment]
@ThangHuuVu ahh this runs over the meta / facebook developer portal it looks like, right? 

I wasn't able to create a Facebook nor Instagram developer app without "verifying my identity" with them by uploading a copy of my passport or other ID, which I definitely did not want Meta having :joy:

**Comment by ThangHuuVu** [comment]
Haha ok no problem, I think we can merge this and I will set it up later 👍

**Review by ThangHuuVu — APPROVED** [review]

File: packages/core/src/providers/threads.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ *     }),
+ *   ],
+ * })
+ * ```
+ *
+ * ### Resources
+ *
+ * - [Threads OAuth documentation](https://developers.facebook.com/docs/threads)
+ * - [Threads OAuth apps](https://developers.facebook.com/apps/)
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:74]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ * ### Resources
+ *
+ * - [Threads OAuth documentation](https://developers.facebook.com/docs/threads)
+ * - [Threads OAuth apps](https://developers.facebook.com/apps/)
+ * 
+ * ### Notes
+ *
+ * :::warning
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:78]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * - [Threads OAuth documentation](https://developers.facebook.com/docs/threads)
+ * - [Threads OAuth apps](https://developers.facebook.com/apps/)
+ * 
+ * ### Notes
+ *
+ * :::warning
+ * 
+ * Email address is not returned by the Threads API.
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:80]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+ * - [Threads OAuth apps](https://developers.facebook.com/apps/)
+ * 
+ * ### Notes
+ *
+ * :::warning
+ * 
+ * Email address is not returned by the Threads API.
+ * 
+ * :::
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:82]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+ * ### Notes
+ *
+ * :::warning
+ * 
+ * Email address is not returned by the Threads API.
+ * 
+ * :::
+ * 
+ * :::tip
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:84]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+ * :::warning
+ * 
+ * Email address is not returned by the Threads API.
+ * 
+ * :::
+ * 
+ * :::tip
+ * 
+ * Threads required callback URL to be configured in your Facebook app and Facebook required you to use **https** even for localhost! In order to do that, you either need to [add an SSL to your localhost](https://www.freecodecamp.org/news/how-to-get-https-working-on-your-local-development-environment-in-5-minutes-7af615770eec/) or use a proxy such as [ngrok](https://ngrok.com/docs).
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:86]:
```suggestion
 *
```

File: packages/core/src/providers/threads.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+export default function Threads(
+  config: OAuthUserConfig<ThreadsProfile>
+): OAuthConfig<ThreadsProfile> {
+  return {
+    id: "threads",
+    name: "Threads",
+    type: "oauth",
+    checks: ["state"],
+    authorization:
+      "https://threads.net/oauth/authorize?scope=threads_basic",
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:118]:
```suggestion
    authorization: "https://threads.net/oauth/authorize?scope=threads_basic",
```

File: packages/core/src/providers/threads.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    token: "https://graph.threads.net/oauth/access_token",
+    userinfo: 
+    "https://graph.threads.net/v1.0/me?fields=id,username,threads_profile_picture_url,threads_biography",
```
Comment (k-taro56) [inline packages/core/src/providers/threads.ts:121]:
```suggestion
    userinfo:
      "https://graph.threads.net/v1.0/me?fields=id,username,threads_profile_picture_url,threads_biography",
```

**Review by k-taro56 — COMMENTED** [review]
Run `pnpm format`

**Comment by k-taro56** [comment]
I'm sorry, I did `Add single comment` by mistake.
I should have done `Start a Review`.

**Comment by ThangHuuVu** [comment]
@k-taro56 no worries, I formatted the file per your suggestion.
@c7e715d1b04b17683718fb1e8944cc28 I leave the biography out of the requesting scope since its not necessary for next-auth, I see that you already written in the doc that it can be added if necessary so people can adjust if needed 👍

**Comment by c7e715d1b04b17683718fb1e8944cc28** [comment]
> @c7e715d1b04b17683718fb1e8944cc28 I leave the biography out of the requesting scope since its not necessary for next-auth, I see that you already written in the doc that it can be added if necessary so people can adjust if needed 👍

We certainly should not include unnecessary parameters in the request.
I forgot about that, thank you for handling it!

File: packages/core/src/providers/threads.ts [hunk 9]
```diff
@@ -58,32 +58,27 @@ export interface ThreadsProfile {
  *
  * const request = new Request(origin)
  * const response = await Auth(request, {
- *   providers: [
- *     Threads({
- *       clientId: THREADS_CLIENT_ID,
- *       clientSecret: THREADS_CLIENT_SECRET,
- *     }),
- *   ],
+ *   providers: [Threads],
```
[the remark this answers is not in the document]
Comment (ThangHuuVu) [reply packages/core/src/providers/threads.ts:61]: Ah I see, the core package is used so there is no env var inference. Let me revert.

File: docs/pages/getting-started/providers/threads.mdx [hunk 10]
```diff
@@ -0,0 +1,83 @@
+import { Callout } from "nextra/components"
+import { Code } from "@/components/Code"
+
+<img align="right" src="/img/providers/threads.svg" height="64" width="64" />
+
+# Instagram Provider
```
Comment (badrange) [inline docs/pages/getting-started/providers/threads.mdx:6]: Is this headline correct or should it have been "Threads Provider"?

Comment (k-taro56) [reply docs/pages/getting-started/providers/threads.mdx:6]: Thank you for pointing this out. I agree with your suggestion.
I have created a new PR.
https://github.com/nextauthjs/next-auth/pull/11245

Comment (badrange) [reply docs/pages/getting-started/providers/threads.mdx:6]: Thanks for responding so quickly!

**Pull request merged**